fe/fe-core/src/main/java/com/starrocks/catalog/FileTable.java

```diff
@@ -0,0 +1,230 @@
+import com.starrocks.thrift.TColumn;
+import com.starrocks.thrift.TFileTable;
+import com.starrocks.thrift.TTableDescriptor;
+import com.starrocks.thrift.TTableType;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class FileTable extends Table {
+    private static final String JSON_KEY_FILE_PATH = "path";
+    private static final String JSON_KEY_FORMAT = "format";
+    private static final String JSON_KEY_FILE_PROPERTIES = "fileProperties";
+
+    private Map<String, String> fileProperties = Maps.newHashMap();
+
+    private List<RemoteFileDesc> fileDescs = null;
+
+    public FileTable() {
+        super(TableType.FILE);
+    }
+
+    public FileTable(long id, String name, List<Column> fullSchema, Map<String, String> properties) throws DdlException {
+        super(id, name, TableType.FILE, fullSchema);
+        this.fileProperties = properties;
+        validate(properties);
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of file table, " + "they are path and format");
+        }
+        Map<String, String> copiedProps = Maps.newHashMap(properties);
+
+        String path = copiedProps.get(JSON_KEY_FILE_PATH);
+        if (Strings.isNullOrEmpty(path)) {
+            throw new DdlException("path is null. Please add properties(path='xxx') when create table");
+        }
+        copiedProps.remove(JSON_KEY_FILE_PATH);
+
+        String format = copiedProps.get(JSON_KEY_FORMAT);
+        if (Strings.isNullOrEmpty(format)) {
+            throw new DdlException("format is null. Please add properties(format='xxx') when create table");
+        }
+        if (!format.equalsIgnoreCase("parquet") && !format.equalsIgnoreCase("orc")) {
+            throw new DdlException("not supported format: " + format);
+        }
+        copiedProps.remove(JSON_KEY_FORMAT);
+
+        if (!copiedProps.isEmpty()) {
+            throw new DdlException("Unknown table properties: " + copiedProps.toString());
+        }
+
+        initFileDescs();
+    }
+
+    public String getTableLocation() {
+        return fileProperties.get(JSON_KEY_FILE_PATH);
+    }
+
+    public RemoteFileInputFormat getFileFormat() {
+        if (fileProperties.get(JSON_KEY_FORMAT).equalsIgnoreCase("parquet")) {
+            return RemoteFileInputFormat.PARQUET;
+        } else if (fileProperties.get(JSON_KEY_FORMAT).equalsIgnoreCase("orc")) {
+            return RemoteFileInputFormat.ORC;
+        } else {
+            return RemoteFileInputFormat.UNKNOWN;
+        }
+    }
+
+    public Map<String, String> getFileProperties() {
+        return fileProperties;
+    }
+
+    public List<RemoteFileDesc> getFileDescs() throws DdlException {
```

Comment:
how to process dir files changes

Reply:
up to now, if dir changed, we need a new filetable.

Reply:
you have cached the file descs. how to process new files in fir